students/bplanica/lesson05/mailroom3.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env python3
+
+# ------------------------------ #
+# Assignment 4 (Mailroom Part 2) for Python 210
+# Dev: Breeanna Planica
+# ChangeLog: (who, when, what)
+#   BPA, 8/11/2019, Created and tested script (Mailroom Part 1)
+#   BPA, 8/18/2019, Updated to add dictionaries and dispatch menu
+#                   Adding function to print emails to text files
+#   BPA, 8/20/2019, Updated dictionary for main donor list to be solely a dictionary
+#   BPA, 8/31/2019, Added error/exception handling
+# ------------------------------ #
+
+# ----- DATA ----- #
+# ---------------- #
+donors = {"bill gates": ("Bill Gates", [326892.24, 326892.24]),
+"mark zuckerberg": ("Mark Zuckerberg", [5465.39, 5465.39, 5465.39]),
+"jeff bezo": ("Jeff Bezo", [877.33]),
+"paul allen": ("Paul Allen", [236.14, 236.14, 236.14]),
+"m bezo": ("M Bezo", [110463.25])} # holds original data set to start with
+
+dict_menu = {} # holds menu options (dispatch)
+str_menu = ("""--------------------
+Choose an action:
+1 - Send a Thank You to a single donor
+2 - Create a Report
+3 - Send letters to all donors
+4 - Quit
+--------------------
+
+""") # holds menu options
+
+
+# ----- PROCESSING ----- #
+# ---------------------- #
+def main():
+    """ main function """
+    while True: # display a menu of choices to the user
+        try:
+            response = int(input(str_menu)) # refers to preset menu declared under Data
+            if response == 4: raise SystemExit
+            dict_menu.get(response)()
+        except TypeError:
+            print("\nPlease choose an action from the above list.")
+            print("Type the menu number associated with the action you would like to perform.")
+        except ValueError:
+            print("\nPlease choose an action from the above list.")
+            print("Type the menu number associated with the action you would like to perform.")
+
+
+def one_thank_you():
+    """ option to add a new entry to the list and print a thank you email"""
+    name = input("Type 'exit' to return the main menu. Who would you like to send a Thank You to?: ").title().strip() # gather name
+    if name.upper() == "EXIT":
+        return # return to main menu
+    try:
+        donation = input(f"Type 'exit' to return the main menu. What was the donation amount for {name:s}?: ").strip() # gather donation
+        print()
+        if donation.upper() == "EXIT":
+            return # return to main menu
+        else:
+            donation = float(donation)
+    except:
+        print("Please enter a valid donation amount. \n")
```

Comment:
This is bouncing the user back to the main menu.  Recommend continuing to ask for a valid amount or allowing user to choose to exit.